File: 30daysOFcode_Java/day 24.java
import java.io.*;
import java.util.*;


class Node{

	int data;
	Node next;

	Node(int d){
        data=d;
        next=null;
    }	
}

class Solution
{

    public static Node removeDuplicates(Node head) {
        //Write your code here
        Node p = head;
        /*
        while (p.next != null) {
            if (p.data == p.next.data)
                p.next = p.next.next;
                
            p = p.next;
        }
        */
  
        while(p!=null)
        { 
            Node q=p.next;
            while(q!=null)
            {
                if(p.data==q.data)
                    p.next=q.next;
                q=q.next;    
            }
            p=p.next;
        }
  
        return head;

    }

    public static  Node insert(Node head,int data){
        Node p=new Node(data);			
        if(head==null)
            head=p;
        else if(head.next==null)
            head.next=p;
        else
        {
            Node start=head;
            while(start.next!=null)
                start=start.next;
            start.next=p;

        }
        return head;
    }

    public static void display(Node head)
    {
        Node start=head;
        while(start!=null)
        {
            System.out.print(start.data+" ");
            start=start.next;
        }
    }
    public static void main(String args[])
    {
        Scanner sc=new Scanner(System.in);
        Node head=null;
        int T=sc.nextInt();
        while(T-->0){
            int ele=sc.nextInt();
            head=insert(head,ele);
        }
        head=removeDuplicates(head);
        display(head);
    }
}

File: 30daysOFcode_Swift/day 24.swift
import Foundation

// Start of class Node
class Node {
    var data: Int
    var next: Node?

    init(d: Int) {
        data = d
    }
} // End of class Node

// Start of class LinkedList
class LinkedList {
    func insert(head: Node?, data: Int) -> Node? {
        if head == nil {
            return Node(d: data)
        }

        head?.next = insert(head: head?.next, data: data)

        return head
    }

    func display(head: Node?) {
        if head != nil {
            print(head!.data, terminator: " ")

            display(head: head?.next)
        }
    }
    // Start of function removeDuplicates
    func removeDuplicates(head: Node?) -> Node? {
        // Add your code here
        var previousNode = head
        var current = head
        while(current != nil) {
            current = current?.next
            if(current?.data == previousNode?.data) {
                previousNode?.next = current?.next
            } else {
                previousNode = current
            }
        }
        return head
    } // End of function removeDuplicates
} // End of class LinkedList

var head: Node?
let linkedList = LinkedList()

let t = Int(readLine()!)!

for _ in 0..<t {
    head = linkedList.insert(head: head, data: Int(readLine()!)!)
}

linkedList.display(head: linkedList.removeDuplicates(head: head))
